src/main/java/net/christophschubert/cp/testcontainers/LdapContainer.java

```diff
@@ -3,92 +3,126 @@
+            builder.append(formatGroupEntry(entry, groupIdNumber++));
         }
 
-        var uidNumber = 10000;
+        var userIdNumber = 10000;
+
         for (Map.Entry<String, String> entry : ldapUsers.entrySet()) {
-            builder.append(formatUserEntry(entry.getKey(), entry.getValue(), uidNumber++));
+            builder.append(formatUserEntry(entry.getKey(), entry.getValue(),
+                    userIdNumber++, members.getOrDefault(entry.getKey(), Set.of())));
         }
 
         return builder.toString();
     }
 
-    static String formatGroupEntry(String groupName, int gidNumber) {
-        final var template = "dn: cn=%s,ou=groups,{{ LDAP_BASE_DN }}\n" +
-                "objectClass: top\n" +
-                "objectClass: posixGroup\n" +
-                "cn: %s\n" +
-                "gidNumber: %d\n\n";
-        return String.format(template, groupName, groupName, gidNumber);
+    private static final String GROUP_TEMPLATE =
+            "dn: cn=%s,ou=groups,{{ LDAP_BASE_DN }}\n" +
+                    "objectClass: top\n" +
+                    "objectClass: posixGroup\n" +
+                    "cn: %s\n" +
+                    "gidNumber: %d\n\n";
+
+    private static String formatGroupEntry(String groupName, int gidNumber) {
+        return String.format(GROUP_TEMPLATE, groupName, groupName, gidNumber);
     }
 
-    static String formatUserEntry(String username, String password, int uidNumber) {
-        final var template = "dn: cn=%s,ou=users,{{ LDAP_BASE_DN }}\n" +
-                "objectClass: inetOrgPerson\n" +
-                "objectClass: posixAccount\n" +
-                "uid: %s\n" +
-                "cn: %s\n" +
-                "sn: Snow\n" +
-                "uidNumber: %d\n" +
-                "gidNumber: 5000\n" +
-                "userPassword: %s\n" +
-                "homeDirectory: /home/%s\n\n";
-        return String.format(template, username, username, username, uidNumber, password, username);
+    private static final String USER_TEMPLATE =
+            "dn: cn=%s,ou=users,{{ LDAP_BASE_DN }}\n" +
+                    "objectClass: inetOrgPerson\n" +
+                    "objectClass: posixAccount\n" +
+                    "%s" +
+                    "uid: %s\n" +
+                    "cn: %s\n" +
+                    "sn: Snow\n" +
+                    "uidNumber: %d\n" +
+                    "gidNumber: 5000\n" +
+                    "userPassword: %s\n" +
+                    "homeDirectory: /home/%s\n\n";
+
+    private static String formatUserEntry(String username, String password, int uidNumber, Set<String> members) {
+        return String.format(USER_TEMPLATE, username, assembleMembers(members),
+                username, username, uidNumber, password, username);
     }
 
-}
+    private static String assembleMembers(Set<String> members) {
+        return members.stream()
+                .map(s -> String.format("memberOf: cn=%s,ou=groups,{{ LDAP_BASE_DN }}\n", s))
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.